### PR TITLE
Rebind pug_html instead of using an arg of the same name

### PR DESCRIPTION
**Pull request by valpackett**
The `function(pug_html)` approach did not work with mixins, since mixins are functions that reference the original pug_html variable from the beginning of template code! Mixins simply returned no output.

This probably needs more testing but seems to work for me.

### CODE REVIEW

File: lib/parse.js [hunk 1]
```diff
@@ -47,11 +47,11 @@ function parse() {
...
 
   const nodes = [
-    Object.assign({}, codeBlockDefaults, { val: 'try { pug_html += function(pug_html){' }),
+    Object.assign({}, codeBlockDefaults, { val: '(function () { var real_pug_html = pug_html; pug_html = ""; try {' }),
     tryBlock,
-    Object.assign({}, codeBlockDefaults, { val: "; return pug_html; }('') } catch (e) {" }),
+    Object.assign({}, codeBlockDefaults, { val: '; real_pug_html += pug_html; } catch (e) { pug_html = ""; ' }),
     catchBlock,
-    Object.assign({}, codeBlockDefaults, { val: ';}' }),
+    Object.assign({}, codeBlockDefaults, { val: '; real_pug_html += pug_html; }; pug_html = real_pug_html; }).bind(this)()' }),
```
Comment (austindebruyn) [inline lib/parse.js]: I'd prefer a `.call(this)` instead of `.bind(this)()`.

**Review by austindebruyn — CHANGES_REQUESTED** [review]
hey thanks for the fix - looks like it should work. The tests all pass, but there's a linting error. I don't use pug mixins so I haven't run into this issue, but I should get some time soon to take a look and get this merged. hopefully we can get a test case in there that simulates mixin usage

**Comment by valpackett** [comment]
Added a test with actual mixin usage.

**Review by austindebruyn — APPROVED** [review]

**Comment by austindebruyn** [comment]
The linting error was actually my fault - master wasn't passing. thanks for the fix :+1:

**Comment by austindebruyn** [comment]
went out as ` pug-plugin-try-catch@1.2.0`

**Pull request merged**